MapboxVisionTests/ObservableVideoSourceTests.swift

```diff
@@ -0,0 +1,130 @@
+@testable import MapboxVision
+import XCTest
+
+private enum Constants {
+    static let numberOfConcurrentMethodCalls = 10000
+    static let expectedTestTimeout = 5.0
+}
+
+class ObservableVideoSourceTests: XCTestCase {
+    private var observableVideoSource: ObservableVideoSource!
+    private var callsCounter: Int!
+
+    private var testExpectation: XCTestExpectation!
+
+    private var concurrentQueue = DispatchQueue(label: "com.mapbox.MapboxVision.ObservableVideoSourceTests.concurrentQueue",
+                                                qos: .default,
+                                                attributes: .concurrent)
+    private var serialQueue = DispatchQueue(label: "com.mapbox.MapboxVision.ObservableVideoSourceTests.serialQueue",
+                                            qos: .default,
+                                            attributes: [])
+
+    override func setUp() {
+        super.setUp()
+        self.observableVideoSource = ObservableVideoSource()
+    }
+
+    // MARK: - Tests
+
+    func testAddMethodDoesNotThrowWhenIsCalledSerially() {
+        // Given state from setUp()
+
+        // When
+        for _ in 1...Constants.numberOfConcurrentMethodCalls {
+            let observer = VideoSourceObserverMock()
+
+            // Then
+            XCTAssertNoThrow(observableVideoSource.add(observer: observer))
+        }
+    }
+
+    func testRemoveMethodDoesNotThrowWhenIsCalledSerially() {
+        // Given state from setUp()
+
+        // When
+        for _ in 1...Constants.numberOfConcurrentMethodCalls {
+            let observer = VideoSourceObserverMock()
+
+            // Then
+            XCTAssertNoThrow(observableVideoSource.add(observer: observer))
```

Comment:
what's the difference with the previous one?

Reply:
Good catch, copy-paste issue, updated.